### PR TITLE
chore: refactor knuu object to use options as argument

### PR DESCRIPTION
**Pull request by mojtaba-esk**
This PR proposes a change to use options as an argument in the new function instead of using WithFuncs. 

## Summary by CodeRabbit

- **Refactor**
  - Improved code readability and maintainability by centralizing configuration options into a single `Options` struct.
  - Replaced individual option functions with a unified `Options` struct for easier management.

- **Tests**
  - Updated test cases to use the new `Options` struct instead of individual options, ensuring consistency and simplifying test setup.

### CODE REVIEW

File: pkg/knuu/knuu.go [hunk 1]
```diff
@@ -236,3 +159,66 @@ func (k *Knuu) handleTimeout(ctx context.Context) error {
 
 	return nil
 }
+
+func loadEnvVariables() error {
+	if err := godotenv.Load(); err != nil && !os.IsNotExist(err) {
+		return ErrCannotLoadEnv.Wrap(err)
+	}
+	logrus.Info("The .env file does not exist, continuing without loading environment variables.")
+	return nil
```
Comment (MSevey) [inline pkg/knuu/knuu.go:168]: won't this always log that the env file doesn't exist even if it does? 
Shouldn't we update to:
```suggestion
	err := godotenv.Load()
	if err != nil && !os.IsNotExist(err) {
		return ErrCannotLoadEnv.Wrap(err)
	}
	if os.IsNotExist(err) {
	        logrus.Info("The .env file does not exist, continuing without loading environment variables.")
	}
	return nil
```

**Review by MSevey — DISMISSED** [review]
Minor comment on the env loading. Fine to do as a follow up though. 

Also let's reference the issue this is closing to make sure it closes it on merge.

File: pkg/knuu/knuu.go (hunk 1 above)
Comment (mojtaba-esk) [reply pkg/knuu/knuu.go:168]: Nice catch!

**Review by MSevey — APPROVED** [review]
Closes #410

**Comment by mojtaba-esk** [comment]
> Closes #410

Thanks for finding it. I did not.

**Pull request merged**